snapcraft/meta/snap_yaml.py

```diff
@@ -558,3 +580,21 @@ def _populate_environment(
         del environment["PATH"]
 
     return environment if environment else None
+
+
+def _process_components(
+    components: Dict[str, models.Component] | None
+) -> Dict[str, ComponentMetadata] | None:
+    """Convert Components from a project to ComponentMetadata for a snap.yaml.
+
+    :param components: Component data from a project model.
+
+    :returns: A dictionary of ComponentMetadata or None if no components are defined.
+    """
+    if not components:
+        return None
+
+    return {
+        name: ComponentMetadata.unmarshal(data.marshal())
```

Comment:
How do you feel about a `from_component` classmethod on `ComponentMetadata` that takes the component itself? The initial implementation could just be `return cls.unmarshal(component.marshal()`, but then if we later need to do anything fancy we don't have to find cases like this.

Reply:
Good idea, that is more readable and usable.

Reply:
yes, this is what I actually wanted :-)